### DISCUSSION TITLE
Order of state changes: event vs. switch

### DISCUSSION THREAD

**Discussion by mfschumann** [SukramJ/aiohomematic#1363] [General]
Hi,

I am using Homematic(IP) local with Home Assistant and would like to set up an automation with HmMIP-BSM switches where the action to be executed depends on both
- which button of the switch was pressed, and how (short/long)
- what state the switch was in before pressing the button 

I thought about triggering on state changes of the event entities, but that will only work reliably if the event entity always changes state before (or after) the switch entity. I did a quick test where the event entity seemed to change first always, but is there any guarantee about the order?

FWIW, this is what I want to achieve in the end: I have three switches that I would like to toggle three lights as shown in this matrix. For example: If Schalter OG is pressed short, it is supposed to toggle Licht OG and set Licht EG to the same state as Licht OG, so I need to know how the switch was pressed (short), and in what state Licht OG was before pressing the switch.

![image](https://github.com/danielperna84/hahomematic/assets/9102962/c078f36f-fa49-4c30-a2f2-c58fa218ce9a)

**Comment by SukramJ** [comment]
> but is there any guarantee about the order?

I think so.
